app/scripts/directives/blockRegistration.js

```diff
@@ -87,6 +87,14 @@ angular
               ) {
                 return;
               }
+              // clear saved value of answer state and zip if address question and country changes
+              if (
+                answer.value.country &&
+                answer.value.country !== oldAnswer.value.country
+              ) {
+                answer.value.state = '';
+                answer.value.zip = '';
+              }
```

Comment:
I really don't like having to do this at all. The current logic didn't do any checks like this, so having to add this for a specific use case feels meh. 

The reason behind needing this is, say I change the country on my address question to 'UZ' from 'US', and then fill in the state input with ```Cool State```. If I change the country back to 'US', the input switches back to a select and it can't find a valid option for ```Cool State``` so it renders blank. But it has a value so from the underlying logic, the form is valid and the user can continue with an invalid US state. Maybe this is an extra edge case, but I think it is one that could occur and needs to be covered. Just wish it could happen some other way. Opinions or suggestions for a different approach are welcomed 🙂

Reply:
One thought I just had would be to have a check when submitting the form that if the address question is required and the country is US, then the state value has to be one of the 50 states. Could just have this be an array and check that the value of state is included in that array.